### ISSUE TITLE
cookie consent rendering issue / how to ignore some script tags on prerendering ?

### ISSUE THREAD

**Issue by darkylmnx** [chrisvfritz/prerender-spa-plugin#247]
I'm having some issue that I can't solve with the current options.

I have a "cookie consent" script, but this scripts renders some UI on the page for the user to consent to the use of cookies, when the page is prerendered, this script doesn't work because it tries to rerender again at runtime.

Same goes for analytics or any script that renders markup on prerender, that then tries to rerender it again at runtime.

I saw a "postPrecess", is there a "preProcess" available to ignore some script tags on prerendering but leave those script tags at runtime ?

What I mean by ignoring them on prerendering, is to simple ignore those scripts when trying to get the markup but then add them back when the page is fully rerendered, so as to have them in the actual html file.

Maybe a comment / prerender / uncomment could do the trick ?

![image](https://user-images.githubusercontent.com/1426357/45063752-f6bac200-b0af-11e8-8a25-36c3a38c4a5d.png)

Here's the actual cookie consent script

**Comment by darkylmnx** [comment]
Same thing for ionicons, there CSS is rendered while this is good at runtime, it's not specially wanted at build time.
![image](https://user-images.githubusercontent.com/1426357/45064790-02a98280-b0b6-11e8-8f63-25215e12811b.png)

**Comment by Tribex** [comment]
@darkylmnx Preprocessing happens in your own code on the page. You can inject a variable during prerendering and only trigger Cookie Consent if the variable is not present.

**Comment by darkylmnx** [comment]
Yes, that's okay for the cookie consent because there's a JavaScript
function call but for ionicons there's just the script tag so that doesn't
work.

I've already figured out the cookie consent with some thing like :
```javascript
If (!window.__PRERENDER_INJECTED.ignore) {
 // Trigger cookie consent
}
```

**Comment by Tribex** [comment]
Not sure the best way to get rid of ionicons without postProcess weirdness.

**Comment by roarkmccolgan** [comment]
+1 for this issue.

@darkylmnx could you let us know how you got around this issue please?
